ui/page/components/utils.go

```diff
@@ -12,3 +17,22 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+// RetryFunc implement retry policy for processes that needs to be executed
+// the after initial failure.
+func RetryFunc(retryAttempts int, sleepDur time.Duration, funcDesc string, errFunc func() error) (int, error) {
+	var err error
+	for i := 0; i < retryAttempts; i++ {
+		if i > 0 {
+			sleepDur *= 2
```

Comment:
Then surround this with an if statement:
```suggestion
			if i > 1 { sleepDur *= 2 }
```